### DISCUSSION TITLE
Add (elementary) inference loops?

### DISCUSSION THREAD

**Discussion by rlouf** [blackjax-devs/blackjax#85] [New features]
`blackjax` purposefully does not provide inference loops at the moment. This could be however useful for two main reasons:

1. These loops are used constantly in integration tests, so this would simplify the code.
2. This would make the library more user-friendly;

We could use [MCX](https://github.com/rlouf/mcx/blob/master/mcx/sample.py)'s inference engine, where we use JAX's host callbacks to add a progress bar within `scan` with very little overhead.

This is an open discussion since this changes slightly the scope of the library.

@junpenglao

**Comment by junpenglao** [comment]
I think that's a good idea - i am not too worry about the scope change, but there might be a discussion to have on how extensive we want the API to be (allow to accept kernel, etc)

**Comment by AdrienCorenflos** [comment]
Maybe a starting point would be to implement it as common utility in the
tests folder and not directly in the public API to see how often it needs
to be modified (as in, can it be stable or is it doomed to change every
time a new method is implemented)?

**Comment by junpenglao** [comment]
🤖 *This reply was drafted with Claude and reviewed by me (@junpenglao).*

Added: `blackjax.util.run_inference_algorithm(rng_key, inference_algorithm, num_steps, initial_state=..., progress_bar=...)`. Closing.
